### DISCUSSION TITLE
导出 org-mode

### DISCUSSION THREAD

**Discussion by Kenshin** [Kenshin/simpread#3775] [插件说明]
描述
---

> 导出当前页面的标注为 `org-mode` 格式，一般针对于 Logseq 的导入方案。

地址
---

[https://simpread.ksria.cn/plugins/details/etyod6A5ZT](https://simpread.ksria.cn/plugins/details/etyod6A5ZT)

快捷键
---

全局快捷键 `o g` → 手动导出标注

权限
---

因为涉及到 **[Markdown 模板辅助增强](https://github.com/Kenshin/simpread/discussions/3725)**，所以此插件为 **高级账户权限** 才能使用。

配置库
---

[简悦 · 配置库](https://github.com/Kenshin/simpread/discussions/4531) 是简悦官方推出的一套针对新用户的极简配置方案，方便新用户用最快的方式使用简悦的各种高级服务，配置库内置了常用的双链笔记用法，如：Notion、Obsidian、Logseq、Roam Research，同时包含了简悦在阅读模式上的一些常规插件：Live Editor、题图、Safari 阅读模式等。

目前配置库包含了 Logseq 的一站式配置方案，细节 [请看这里](https://github.com/Kenshin/simpread/discussions/4531#discussioncomment-3844896)。

教程
--- 

一站式教程 [请看这里](https://zhuanlan.zhihu.com/p/505863844)。

自动化
---

1. 利用 [自动化辅助增强](https://github.com/Kenshin/simpread/discussions/3596) + [Markdown 模板辅助增强](https://github.com/Kenshin/simpread/discussions/3725) + [此模板](https://github.com/Kenshin/simpread/discussions/2153#discussioncomment-2627320) 可以实现：当稍后读、标注变化时自动生成标注文件

2. 利用 Hazel 将标注产生的 `.org` 自动导入到 Logseq 的目录。

上述配置 [请看这里](https://github.com/Kenshin/simpread/discussions/3775#discussioncomment-2643234)。

视频
---

https://user-images.githubusercontent.com/81074/165430697-b22a294d-676b-4c32-a2e4-f7db7b23b9b5.mp4

截图
---

![image](https://user-images.githubusercontent.com/81074/165220748-ddcc6b25-0f10-415a-a728-2de6f4fcea10.png)

选项
---

> 随便进入一个页面 e.g. https://sspai.com/post/71576 并进入 **阅读模式 → 右下角触发器 → 动作 → 插件触发器**

按照下图所示即可查看此插件的选项页。（注意：如果你的界面没有插件触发器，请根据此 [解决方案](https://github.com/Kenshin/simpread/discussions/2342) 修复。

![image](https://user-images.githubusercontent.com/81074/165056218-b0c977b6-4380-4d70-84ed-5c22993d6c88.png)

功能说明
---

1. 定制导出文件名

   支持占位符，详细说明 [请看这里](https://github.com/Kenshin/simpread/discussions/3775#discussioncomment-2613468)。

   ![image](https://user-images.githubusercontent.com/81074/165071299-9713af43-a706-47ed-a7e8-8be05ed4a2a2.png)

2. 导出方式

   包括：通过浏览器导出（到浏览器默认的下载目录）以及通过同步助手导出到 `output` 一般来说使用此功能的用户都是选择同步助手导出方案。

    ![image](https://user-images.githubusercontent.com/81074/165071907-8f2882ca-8034-42ed-ac3f-c6ac203dbbee.png)
    
3. 目前只能开启 [Markdown 模板辅助增强](https://github.com/Kenshin/simpread/discussions/3725) 后才能使用此功能。

    ![image](https://user-images.githubusercontent.com/81074/165073027-011b550b-dcf6-44ee-923a-5680547894a0.png)

更新日志
---

### 1.0.2

- `Fix` 当标题出现特殊字符时无法使用的错误。

### 1.0.1

- `Optimize` 自动化触发时的触发逻辑。

### 1.0.0

- 手动导出，支持快捷键 `o g`

- 自动化导出

- 选项

    - 自定义标题

    - 导出方案，包括： `同步助手` 与 `浏览器`

    - 使用 [Markdown 模板辅助增强](https://github.com/Kenshin/simpread/discussions/3725)

- 权限验证

引申
---

> 安装插件或带有插件触发器的插件失效的 [解决方案](https://github.com/Kenshin/simpread/discussions/2342)。

**Comment by Kenshin** [comment]
# 自定义标题

一般情况下不用修改自定义标题的默认值，因为默认值跟 [知识库](https://kb.simpread.pro/#/page/%E5%BB%BA%E7%AB%8B%E7%9F%A5%E8%AF%86%E5%BA%93) 是一个命名体系。

| 占位符             | 描述                                                                                     |
|-----------------|----------------------------------------------------------------------------------------|
| `{{id}}`        | 稍后读 ID                                                                                 |
| `{{title}}`     | 稍后读的标题                                                                                 |
| `{{un_title}}`  | 稍后读的标题                                                                                 |
| `{{title}}`     | 原页面的标题                                                                                 |
| `{{timestamp}}` | 当前稍后读建立时的时间戳 e.g. 20212809162353                                                       |
| `{{now\|<格式化>}}` | [详细说明](https://github.com/Kenshin/simpread/discussions/3725#discussioncomment-2577619) |
| `{{note}}`      | 当前稍后读的备注，当不存在时使用稍后读标题，此功能可以理解为别名。                                                                               |
| `{{mode}}`     | 默认值为 `@annote`，一般不用设定                                                                                 |

**Comment by Kenshin** [comment]
# 自动化方案

与 [导出到 Obsidian](https://github.com/Kenshin/simpread/discussions/2902) 插件方案一致。

```
o m u n
o m u u
o m a a
o m a c
```

将上面的自动化快捷键依次加入到 [自动化辅助增强](https://github.com/Kenshin/simpread/discussions/3596) 插件，如下图所示

![image](https://user-images.githubusercontent.com/81074/165056731-7f940465-027e-43af-86ec-7fc2f314659f.png)

## 模板

安装 [Markdown 模板辅助增强](https://github.com/Kenshin/simpread/discussions/3725)，复制 [此模板](https://github.com/Kenshin/simpread/discussions/2153#discussioncomment-2627320) 的内容到下图。

![image](https://user-images.githubusercontent.com/81074/165434493-04bd37ff-21ce-4074-ab56-ba3faaa7b145.png)

需要开启下图所示的选项才可使用

![image](https://user-images.githubusercontent.com/81074/165034089-a8e2987e-7947-4bdc-89e4-5d743fff1f47.png)

## Hazel 设置

可以利用 Hazel、Keyboard Maestro、DropIt 监控 `output` 里面的 `xxx.org` 然后转移到 Logseq 目录。

下图以 Hazel 为例说明

![image](https://user-images.githubusercontent.com/81074/165431275-700a5fd6-b669-42f4-8b5c-8d76ab02faf2.png)

## 视频

https://user-images.githubusercontent.com/81074/165430697-b22a294d-676b-4c32-a2e4-f7db7b23b9b5.mp4

**Comment by jon9527** [comment]
开发者我在这一步怎么没看到.org文件导出？插件都安装了。要同步助手配置的吗？

详细讨论 https://github.com/Kenshin/simpread/issues/4364
